llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -1534,6 +1534,13 @@ RISCVTTIImpl::getArithmeticReductionCost(unsigned Opcode, VectorType *Ty,
              getRISCVInstructionCost(Opcodes, LT.second, CostKind) +
              getCmpSelInstrCost(Instruction::ICmp, ElementTy, ElementTy,
                                 CmpInst::ICMP_EQ, CostKind);
+    } else if (ISD == ISD::XOR) {
+      // Example sequences:
+      //   vsetvli a0, zero, e8, mf8, ta, ma
+      //   vcpop.m a0, v0
+      //   andi a0, a0, 1
+      Opcodes = {RISCV::VCPOP_M};
+      return LT.first + getRISCVInstructionCost(Opcodes, LT.second, CostKind);
```

Comment:
Shouldn't this be
```suggestion
      return LT.first * getRISCVInstructionCost(Opcodes, LT.second, CostKind);
```

With that said, I'm also not sure why the existing sequences are adding the legalization cost instead of multiplying it. We multiply it everywhere else in RISCVTargetTransformInfo. Maybe this is something to fix in a follow up patch

Reply:
OK, removed and will create new patch to deal with it

Reply:
deal with the type: ElementTy->isIntegerTy(1), should we real need use *???? I don't understand  the case of `ISD == ISD::AND`,  it uses `LT.first - 1`. I'm not sure why it subtracts 1.

Reply:
mycost : LT.first  - 1 + getRISCVInstructionCost + 1(1 is cost of the andi)

Reply:
`(LT.first - 1)` first appeared in this [commit](https://github.com/llvm/llvm-project/commit/536095a27c17e2f30765111f57d36c0ee9e211c7).
It seems to be used to calculate the additional cost when the LMUL exceeds 8.
For example,
```
define zeroext i1 @vreduce_and_nxv1i1(<vscale x 128 x i1> %v) {
;    vmand.mm    v8, v0, v8 // addtional cost when LMUL exceeds 8.
;    vmnot.m v8, v8
;    vcpop.m a0, v8
;    seqz    a0, a0
  %red = call i1 @llvm.vector.reduce.and.nxv128i1(<vscale x 128 x i1> %v)
  ret i1 %red
}
```

Reply:
thanks~

Reply:
Yeah LT.first is the cost of type legalization, i.e. splitting. But it's the number of times the operation will be split, so we need to multiply that by the cost of whatever node. I'll open up a PR to fix the existing cases, but for this PR we might as well do it the correct way i.e. by multiplying. (Although I don't think it will make a difference to the costs, since these mask instructions are always LMUL 1!)

Reply:
I've gone ahead and done this in df10f1c6f79a77e45e44c62974a186decb0f1f33

Reply:
Thanks!